docs/guide/features.md

```diff
@@ -368,6 +368,36 @@ By default, the worker script will be emitted as a separate chunk in the product
 import MyWorker from './worker?worker&inline'
 ```
 
+A web worker script can be imported by `new Worker()`. The difference between this syntax and imported by appending `?worker` or `?sharedworker` is that this one can use worker options. So we can create a `classic worker` by this syntax.
+
+Detect the case where `{ type: 'classic' }` is not provided or `{ type: 'classic' }` is passed with a static string and use classic workers.
+
+```ts
+const worker = new Worker(new URL('./worker.js', import.meta.url), {
+  type: 'classic'
+})
+const worker = new SharedWorker(new URL('./worker.js', import.meta.url), {
+  type: 'classic'
+})
+```
+
+::: tip Note
+
+If a dynamic expression appears, vite will issue an error. And give `/* @vite-ignore */` out of Vite processing. Because use dynamic expression, vite will copy the code of `/@vite/env` to the worker code head.
+
+```ts
+const generateWorkerOption = () => ({ type: 'classic' })
+// use dy dynamic expression will got a vite process error.
+const worker = new Worker(
+  new Url('./worker.js', import.meta.url),
+  generateWorkerOption()
+)
+```
```

Comment:
I think we can simplify this part with "the worker options must be static for Vite to analyze it. if it contains dynamic expressions, `/* @vite-ignore */` can be used to opt-out of Vite processing". 

I don't think we need to explain `/@vite/env` as that's an implementation detail. We can also simplify the code block with `{ type: workerType }` (uses variables) perhaps and show how to use the `/* @vite-ignore */` comment. I think `Url` should be `URL` too.